stories/topics/proc-aap-gcp-application-upgrade.adoc

```diff
@@ -0,0 +1,88 @@
+
+Technical Preview
+~~~~~~~~~~~~~~~~~
+
+You can upgrade your existing {AAPonGCP} deployment to the newer version. The upgrade process covers upgrading of automation hub and automation controller. The upgrade process takes roughly the same amount of time as a {AAPonGCP} deployment install. Follow the steps below to ensure a smooth upgrade process.
+
+[NOTE]
+=====
+{AAPonGCP} supports sequential upgrades. All upgrades should be no more than one major version behind what you are currently upgrading to. For example, in order to upgrade to {AAPonGCP} to 2.2.20230215, you must be on the first version of {AAPonGCP}.
+=====
+
+.Prerequisites
+* You require a docker installed in order to run the upgrade playbook.
+
+== Upgrade Process
+
+The following instructions describe how to run the upgrade playbook.
+
+.Procedure
+. Gather the following input variables to create an `extra_vars.yaml` file.
+.. *gcp_service_account_credentials_json_path*: Service account crendential json file path for the google cloud account.
+.. *gcp_deployment_name*: Deployment name which you want to backup.
+.. *gcp_compute_region*: GCP region where foundation deployment deployed.
+.. *gcp_compute_zone*: GCP zone where foundation deployment deployed.
++
+[source,bash]
+----
+gcp_deployment_name: "" 
+gcp_compute_region: ""
+gcp_compute_zone: ""
+gcp_service_account_credentials_json_path: "/secret/<service-account-creds-json-filename>"
+----
++
+. Pull the docker image for "ansible-on-clouds-ops" container having tag with the version which you want to upgrade to.
++
+[NOTE]
+=====
+The "ansible-on-clouds-ops" container image tag must match with the version which you want to upgrade to. For example, if your foundation deployment version is 2.2.0, pull "ansible-on-clouds-ops" image with tag 2.2.1 to upgrade to version 2.2.1.
+=====
++
+[source,bash]
+----
+$ export IMAGE=quay.io/ansible-on-clouds/ansible-on-clouds-ops:2.2.1
+$ docker pull $IMAGE
+----
++
+. Running the upgrade playbook as a container.
++
+.. Set variable `GOOGLE_APPLICATION_CREDENTIALS` to the google cloud credential Json file path.
++
+[source,bash]
+----
+$ export GOOGLE_APPLICATION_CREDENTIALS=<path to the google credential Json file>
+----
++
+.. Replace `<absolute path to google application credentials dir>` with your path to the service account credential Json file for google cloud and `<absolute path of extra vars dir>` with the path to your directory where you `extra_vars.yaml` file exists which you have created in step 1 of upgrade process.
++
+Replace `<deployment-name>` with your foundation deployment name.
++
+.. Use the following command to run the playbook.
++
+[source,bash]
+----
+$ docker run --rm \
+  --env PLATFORM=GCP \
+  --env CLOUDSDK_CORE_PROJECT=gc-ansible-cloud \
+  --env GOOGLE_APPLICATION_CREDENTIALS=$GOOGLE_APPLICATION_CREDENTIALS \
+  --env GENERATE_INVENTORY=true \
+  --env DEPLOYMENT_NAME=<deployment-name> \
+  -v <absolute path to google application credentials dir>:/secrets \
```

Comment:
@leena-jawale: 
I had to update SELinux context with the below command in order for container to access file contents in my host directory. Its good to add a note

Error
```
/secrets/gc-ansible-cloud-2f6d0e5e24cb.json does not exist, maybe the directory is not mounted
ERROR: (gcloud.auth.login) Failed to load YAML from [/secrets/gc-ansible-cloud-2f6d0e5e24cb.json]: Unable to read file [/secrets/gc-ansible-cloud-2f6d0e5e24cb.json]: [Errno 13] Permission denied: '/secrets/gc-ansible-cloud-2f6d0e5e24cb.json'
```

workaround that helped me
```
old SE-Linux context: unconfined_u:object_r:user_home_t:s0

[pkesavar@pkesavar secret]$ chcon -R -t container_file_t /home/pkesavar/secret

[pkesavar@pkesavar secret]$ ls -lZ /home/pkesavar/secret/
total 4
-rwxrwxrwx. 1 pkesavar pkesavar unconfined_u:object_r:container_file_t:s0 2315 Mar  8 23:02 gc-ansible-cloud-2f6d0e5e24cb.json
[pkesavar@pkesavar secret]$
```

Reply:
For all volume mapping, would be great to add the suffix for access permission.

I tested on Fedora 37 and both volume mapping worked with the `:ro`.